### PR TITLE
CASEFLOW-1119 Create new appellant-substitute appeal matching original appeal

### PR DESCRIPTION
**Pull request by yoomlam**
Resolves [CASEFLOW-1119](https://vajira.max.gov/browse/CASEFLOW-1119)

Controller creates new appeal stream linked to orig appeal and having same status (eg, AOD, CAVC)

### Description
Add test for CAVC status.

### Acceptance Criteria
- [ ] add test for CAVC & Implementation
- [ ] Verify new appeal maintains original appeal status (ex. AOD, CAVC)

### Testing Plan
In the UI,
1. Find or create a CAVC Remand appeal (see updated RSpec)
2. Dispatch the appeal (if it's not already dispatched) so that "Create Substitution" button appears as the COB user.
3. Walk-through the Appellant Substitution workflow
4. Check that the resulting appeal is also CAVC

Repeat the process for an AOD appeal and CAVC+AOD appeal.

Example:
```ruby
a=FactoryBot.create(:appeal, :dispatched, :type_cavc_remand, :advanced_on_docket_due_to_motion)
a.uuid
=> "aef2ada7-d71a-4235-b96a-4d11174da27c"
```
Then go to http://localhost:3000/queue/appeals/aef2ada7-d71a-4235-b96a-4d11174da27c and Create Substitution.

### Screenshot
http://localhost:3000/queue/appeals/ec1a8c45-9978-40bc-a3af-2b6f22edbf4d

image

### CODE REVIEW

File: app/models/appeal.rb [hunk 1]
```diff
@@ -349,7 +349,12 @@ def cavc
   def cavc_remand
     return nil if !cavc?
 
-    CavcRemand.find_by(remand_appeal: self)
+    # If this appeal is a direct result of a CavcRemand, then return it
+    return CavcRemand.find_by(remand_appeal: self) if CavcRemand.find_by(remand_appeal: self)
+
+    # If this appeal went through appellant_substitution after a CavcRemand, then use the source_appeal,
+    # which is the same stream_type (cavc? == true) as this appeal.
+    appellant_substitution.source_appeal.cavc_remand if appellant_substitution?
```
Comment (jcq) [inline app/models/appeal.rb:357]: Just to confirm, would we need to do something similar if there was a docket switch after a CAVC remand? Go back to the source appeal?

File: app/models/appeal.rb [hunk 2]
```diff
@@ -349,7 +349,12 @@ def cavc
   def cavc_remand
     return nil if !cavc?
 
-    CavcRemand.find_by(remand_appeal: self)
+    # If this appeal is a direct result of a CavcRemand, then return it
+    return CavcRemand.find_by(remand_appeal: self) if CavcRemand.find_by(remand_appeal: self)
```
Comment (jcq) [inline app/models/appeal.rb:353]: Is Rails smart enough to cache this so as to not trigger multiple queries? (I assume yes...?)

File: spec/models/appellant_substitution_spec.rb [hunk 3]
```diff
@@ -73,6 +75,34 @@
...
+        let(:source_appeal) { create(:appeal, :dispatched, :type_cavc_remand) }
+        it "copies CAVC status to new appeal" do
+          expect(source_appeal.cavc?).to be true
+
+          appellant_substitution = subject
+          # Source appeal's CAVC status is unchanged
+          expect(source_appeal.cavc?).to be true
+
+          target_appeal = appellant_substitution.target_appeal
+          expect(target_appeal.cavc?).to be true
```
Comment (jcq) [inline spec/models/appellant_substitution_spec.rb:88]: Curious about how `court_remand?` (called by `cavc?`) works, but obviously it does, given this test. 😅

File: spec/models/appellant_substitution_spec.rb [hunk 4]
```diff
@@ -73,6 +75,34 @@
...
+          expect(target_appeal.aod?).to be true
+        end
+      end
```
Comment (jcq) [inline spec/models/appellant_substitution_spec.rb:105]: Good coverage

**Review by jcq — APPROVED** [review]
A couple of (most likely non-blocking) questions, but LGTM!

File: app/models/appeal.rb (hunk 2 above)
Comment (yoomlam) [reply app/models/appeal.rb:353]: Yes

File: app/models/appeal.rb (hunk 1 above)
Comment (yoomlam) [reply app/models/appeal.rb:357]: Probably. If so, we can make this more generic like `some_record.try(: source_appeal)`

File: spec/models/appellant_substitution_spec.rb (hunk 3 above)
Comment (yoomlam) [reply spec/models/appellant_substitution_spec.rb:88]: You mean Rails enums? https://api.rubyonrails.org/v5.2.5/classes/ActiveRecord/Enum.html

File: app/models/appeal.rb (hunk 1 above)
Comment (jcq) [reply app/models/appeal.rb:357]: Would it make sense to do that as a follow-up PR that adds spec coverage for the DS case?

Comment (yoomlam) [reply app/models/appeal.rb:357]: Yes. Definitely not with this PR. You can write up a ticket and assign it to Foxtrot?

**Pull request merged**